[18.0] [MIG] base_multi_company: Migration to 18.0

PieterPaulussen opened the pull request.

lasley committed:
[ADD] base_multi_company: Create new module
* Create new module to provide base multi company logic and mixin
* Add deactivation by company mixin
* Add company_id/ids handling
* Add break after company is found
Squashed commits:
[854cc36] Increase test coverage
[770bd71] Revert hook view create back to model init
[40e803e] Fix apples and oranges
[7a4dfb4] Use registry correctly
[6e9f170] Switch company_id to computed & move company aliased view creation to post init hook
[faa4fc9] Remove active functionality
[fecfb59] Add explicit tests for active and inactive searches

lmignon committed:
10.0 base multi company lmi (#1)

* Revert "Revert hook view create back to model init"

This reverts commit 770bd71a919e5801702f0fd9d398a37c08472a70.

* [FIX] base_multi_company: Always create the view into a pre_init_hook to avoid error in log

lasley committed:
PR fixes (+1 squashed commit)
Squashed commits:
[4c17d04] auto_join company_ids

lmignon committed:
[FIX] base_multi_company: Allow child_of/parent_of operator into search domain on company_id/company_ids (+1 squashed commit)
Squashed commits:
[fe161fe] [ADD] setup.py

Commit:
- Remove depends as company_id is not stored anymore
- Test if a company is set in inverse method (+1 squashed commit)
Squashed commits:
[d670f30] [FIX] fix init hooks as company_id is not stored anymore

oca-transbot committed:
OCA Transbot updated translations from Transifex

lasley committed:
[IMP] base_multi_company: Improve ReadMe
* Add implementation instructions to ReadMe

pedrobaeza committed:
[MIG] *_multi_company: OpenUpgrade migration scripts

acsonefho committed:
Edit unit test + fix bug

[FIX] Fix issue based on the computation of company_id for multi_company_abstract models

[FIX] Fix issue based on the computation of company_id for multi_company_abstract models

[FIX] Fix issue based on the computation of company_id for multi_company_abstract models

[FIX] Bad branch version pushed before

[FIX] Bad branch version pushed before

[FIX] manifest

Fix flake

Flake8 on OCA tests

SimoRubi committed:
[FIX] Insert in company_ids relation's table only ids of records that are bound to some company

oca-travis committed:
[UPD] Update base_multi_company.pot

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 88.9% (8 of 9 strings)

Translation: multi-company-10.0/multi-company-10.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-10-0/multi-company-10-0-base_multi_company/pt/

rodrigets committed:
[MIG] base_multi_company: Migration to 11.0

oca-travis committed:
[UPD] Update base_multi_company.pot

pedrobaeza committed:
[FIX] base_multi_company: Fix tests

Odoo now checks if the user who creates the transient model is the same that is accessing
its data, so we need to create it with the same user.

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

pedrobaeza committed:
[FIX] base_multi_company: Fix several things:

* Abstract model shouldn't have default for `company_ids`. It will be got by
  default `company_id` value.
* Record rule based on `company_id` makes the evaluation slower. Switch to `company_ids` field.
* Add `@api.depends` to `company_id` computation for proper refresh.
* Fix tests that were not correct but luckily previously suceeded. Now that other code has
  been removed, they have been uncovered.

oca-travis committed:
[UPD] Update base_multi_company.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

pedrobaeza committed:
[MIG] partner_multi_company: Finish migration to 11.0

oca-travis committed:
[UPD] Update base_multi_company.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

florian-dacosta committed:
Migrate base_multi_company to v12

florian-dacosta committed:
Improve performance adding a techical field used in ir rules

florian-dacosta committed:
fixup! Migrate base_multi_company to v12

oca-travis committed:
[UPD] Update base_multi_company.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-12.0/multi-company-12.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-base_multi_company/

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 70.0% (7 of 10 strings)

Translation: multi-company-12.0/multi-company-12.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-base_multi_company/zh_CN/

max3903 committed:
[FIX] base_multi_company

OCA-git-bot committed:
[UPD] README.rst

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (10 of 10 strings)

Translation: multi-company-12.0/multi-company-12.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-base_multi_company/hr/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (10 of 10 strings)

Translation: multi-company-12.0/multi-company-12.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-base_multi_company/pt_BR/

alvarorib committed:
Translated using Weblate (Portuguese)

Currently translated at 90.0% (9 of 10 strings)

Translation: multi-company-12.0/multi-company-12.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-12-0/multi-company-12-0-base_multi_company/pt/

osi-scampbell committed:
[IMP] Turn Off IR Rule For Test

osi-scampbell committed:
[IMP] Flake8

JoanSForgeFlow committed:
[IMP] base_multi_company: black, isort

JoanSForgeFlow committed:
[MIG] base_multi_company: Migration to 13.0

oca-travis committed:
[UPD] Update base_multi_company.pot

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] base_multi_company: Pre-commit does this

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-13.0/multi-company-13.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-base_multi_company/

LoisRForgeFlow committed:
[13.0][FIX] base_multi_company: incompatibility error

if you have several companies active and any of them with
lower id than the current company, you can get multi-company
inconsistency errors.

E.g: when reserving stock pickings, product company is computed
and might be different than the move's company.

OCA-git-bot committed:
base_multi_company 13.0.1.0.1

LoisRForgeFlow committed:
[13.0][FIX] base_multi_company: consider force_company context key.

This is needed for transactions in which the company is forced, e.g
intercompany PO/SO synchronization.

OCA-git-bot committed:
base_multi_company 13.0.1.0.2

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (6 of 6 strings)

Translation: multi-company-13.0/multi-company-13.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-base_multi_company/es_AR/

LoisRForgeFlow committed:
[13.0][FIX] base_multi_company: name_search

In some situations the 'in' operator is used with company_id in a
name_search. ORM does not convert to a proper WHERE clause when using
the 'in' operator.
e.g: ```
     WHERE "res_partner"."id" in (SELECT "res_partner_id"
     FROM "res_company_res_partner_rel" WHERE "res_company_id" IN (False, 1)
```
patching the args to expand the cumbersome args int a OR clause fix
the issue.
e.g: ```
     WHERE "res_partner"."id" not in (SELECT "res_partner_id"
             FROM "res_company_res_partner_rel"
             where "res_partner_id" is not null)
         OR  ("res_partner"."id" in (SELECT "res_partner_id"
             FROM "res_company_res_partner_rel" WHERE "res_company_id" IN 1)
```

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: multi-company-13.0/multi-company-13.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-13-0/multi-company-13-0-base_multi_company/es/

rousseldenis committed:
[IMP] base_multi_company: black, isort, prettier

rousseldenis committed:
[14.0][MIG] base_multi_company

Use odoo-test-helper to test Abstract class

rousseldenis committed:
[14.0][IMP] Add test for name_search()

rousseldenis committed:
[14.0][FIX] base_multi_company: Use [] for domain on company_ids

rousseldenis committed:
[14.0][FIX] base_multi_company: Restore False for searches

Odoo seems to have a problem with subqueries, to avoid problems
with False values, store it in a Boolean field

rousseldenis committed:
[14.0][IMP] base_multi_company: Add hook for new field

rousseldenis committed:
[14.0][FIX] base_multi_company: Compute no_company_ids as sudo

rousseldenis committed:
[14.0][IMP] base_multi_company: Avoid warnings in logs

pedrobaeza committed:
[FIX] base_multi_company: Make company_id writeable again

Not being writeable, we have warnings on constraints over this field
(it's used for example in `sale` module for `product.template` model).

Giving an inverse to the field, we avoid this problem and give
compatibility to any other module that writes this field.

rousseldenis committed:
[14.0][IMP] base_multi_company: Allow non-aware multi-company behaviours

rousseldenis committed:
[14.0][IMP] base_multi_company: Add depends_context for env.company

rousseldenis committed:
[14.0][IMP] base_multi_company: Avoid one recordset update + add tests

rousseldenis committed:
[14.0][IMP] base_multi_company: Improve module README

oca-travis committed:
[UPD] Update base_multi_company.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
base_multi_company 14.0.1.1.1

JasminSForgeFlow committed:
[IMP] base_multi_company: black, isort, prettier

JasminSForgeFlow committed:
[MIG] base_multi_company: Migration to 15.0

Commit:
[UPD] Update base_multi_company.pot

OCA-git-bot committed:
[UPD] README.rst

LoisRForgeFlow committed:
[FIX] base_multi_company: remove external dependency

it is only needed for tests.

OCA-git-bot committed:
base_multi_company 15.0.1.0.1

prabakaran04 committed:
[MIG] base_multi_company: Migration to 16.0

pedrobaeza committed:
[FIX] base_multi_company: Proper _check_company when multiple companies selected

Steps to reproduce:

- Create two companies C1 and C2.
- Create a product with C1 and C2 as allowed companies.
- Having C1 as main company, update quantity on hand to 100.
- Switch to C2 as main company, but check the C1 company to be seen as
  well.
- Now update quantity on hand to any other value for the entry of the
  company C1.

Result:

'Stock>Inventory adjustment' belongs to company 'C2' and 'Product'
(product_id: 'Product 1') belongs to another company.

Explanation:

This is because the computed field `company_id` that is used for
checking the product company vs the stock.move company (the error is not
accurate, as it says the location, but it's the move), is returning the
main company where the user is.

Solution:

Inject a context whenever we run `_check_company` for having available
the company involved, and use that company as value for the record being
checked if included in the allowed companies.

renda-dev committed:
[FIX] base_multi_company: apply '_name_search' fix to 'search_read'

Commit:
[UPD] Update base_multi_company.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-16.0/multi-company-16.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-base_multi_company/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-16.0/multi-company-16.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-base_multi_company/

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: multi-company-16.0/multi-company-16.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-base_multi_company/es/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (4 of 4 strings)

Translation: multi-company-16.0/multi-company-16.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-base_multi_company/es_AR/

SirAionTech committed:
[FIX] base_multi_company: Check multiple companies

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: multi-company-16.0/multi-company-16.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-base_multi_company/it/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (4 of 4 strings)

Translation: multi-company-16.0/multi-company-16.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-16-0/multi-company-16-0-base_multi_company/pt_BR/

rven committed:
[IMP] base_multi_company: pre-commit stuff

rven committed:
[MIG] base_multi_company: Migration to 17.0

Commit:
[UPD] Update base_multi_company.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: multi-company-17.0/multi-company-17.0-base_multi_company
Translate-URL: https://translation.odoo-community.org/projects/multi-company-17-0/multi-company-17-0-base_multi_company/

chienandalu committed:
[FIX] base_multi_company: search with in operator

When searching the company with a domain like [("company_id", "in", [1,
False]) to include records which are shared between companies we won't
get those with no companies at all. That will lead to logical errors in
several workflows.

TT51779

[FIX] base_multi_company: Applying the correct parameters in the search_read() method

OCA-git-bot committed:
[BOT] post-merge updates

yajo committed:
[FIX] base_multi_company: support `in False` search operators

Before this commit, when doing a read with a domain such as [the one used by upstream `base.res_partner_rule`][1], the read failed to find records without company.

This is now fixed in the base module, adding tested support for such operators. The only relevant part of the hooks that were provided to workaround the issue is extracted to a new `post_init_hook`. All other hooks are marked as deprecated.

[1]: https://github.com/odoo/odoo/blob/db072461cddced2a8f65a64fb6d2ddf0dd79b38e/odoo/addons/base/security/base_security.xml#L23

@moduon MT-8863 MT-8873

PieterPaulussen committed:
[MIG] base_multi_company: Migration to 18.0

pedrobaeza commented:
/ocabot migration base_multi_company

LoisRForgeFlow reviewed (changes requested):
Hi @PieterPaulussen 

Could you include this change done in 16.0 https://github.com/OCA/multi-company/pull/760 ?

Thanks!

yajo committed:
[FIX] base_multi_company: don't eliminate company_id value

Some forms can send both values, but in that case the one from company_id shouldn't be just dropped. It should be kept instead.

@moduon MT-8873

PieterPaulussen commented:
@LoisRForgeFlow I made an attempt to cherry pick the fix. Let's see if it checks out.

LoisRForgeFlow commented:
@PieterPaulussen it looks like the cherry-pick is correctly executed, however the tests are now red. Could you check?

gbrito commented on base_multi_company/models/multi_company_abstract.py:
This was made redundant from the last PR

```suggestion
```

gbrito commented on base_multi_company/tests/test_multi_company_abstract.py:
We can also remove this test
```suggestion
```

gbrito reviewed (changes requested):
I didn't look if there is another module that depend on that method..
Tests failed because of the deprecation warning

ArnauCForgeFlow commented:
@PieterPaulussen Hello!, any update on the status of this PR?

LoisRForgeFlow commented:
Closed in favor of https://github.com/OCA/multi-company/pull/866
